### PR TITLE
Enable users to pass Labels through Provisioner API

### PR DESCRIPTION
**Pull request by ellistarn**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/controllers/provisioning/v1alpha1/allocator/controller.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func (c *Controller) Reconcile(object controllers.Object) error {
 
 	// 4. Bind pods to nodes
 	for _, packing := range packings {
-		zap.S().Infof("Binding %d pods to node %s", len(pods), packing.Node.Name)
+		zap.S().Infof("Binding %d pods to node %s", len(packing.Pods), packing.Node.Name)
```
Comment (njtran) [inline pkg/controllers/provisioning/v1alpha1/allocator/controller.go:97]: nice!

File: pkg/controllers/provisioning/v1alpha1/allocator/bind.go [hunk 2]
```diff
@@ -54,8 +52,12 @@ func (b *Binder) Bind(ctx context.Context, provisioner *v1alpha1.Provisioner, no
 }
 
 func (b *Binder) decorate(provisioner *v1alpha1.Provisioner, node *v1.Node) {
-	node.ObjectMeta.Labels = map[string]string{
-		KarpenterNodeLabelKey: fmt.Sprintf("%s-%s", provisioner.Name, provisioner.Namespace),
+	node.Labels = map[string]string{
```
Comment (njtran) [inline pkg/controllers/provisioning/v1alpha1/allocator/bind.go:55]: does this need to be `node.ObjectMeta.Labels`?

**Review by njtran — COMMENTED** [review]
LGTM

**Review by njtran — APPROVED** [review]

File: pkg/controllers/provisioning/v1alpha1/allocator/bind.go (hunk 2 above)
Comment (ellistarn) [reply pkg/controllers/provisioning/v1alpha1/allocator/bind.go:55]: It's using embedded struct field inference. They're the same.

**Pull request merged**